stdlib/public/core/Sequence.swift

```diff
@@ -988,7 +988,10 @@ extension Sequence {
         ringBuffer.append(element)
       } else {
         ringBuffer[i] = element
-        i = (i + 1) % maxLength
+        i += 1
+        if i == maxLength {
```

Comment:
not that it is functionally any different (even as a lowered instruction support); I wonder if `i >= maxLength` might be more "defensive".

It might also be good to verify that the compiler isn't optimizing this path for architectures known to have either slow or non-existent division.

Reply:
The verification part I did already.

Reply:
Ok I updated it!